src/main/java/de/muenchen/isi/domain/service/AbfrageStatusService.java

```diff
@@ -302,4 +299,89 @@ public UUID getAbfrageId(final MessageHeaders messageHeaders) throws EntityNotFo
         }
         return abfrageId;
     }
+
+    /**
+     * Entnimmt alle möglichen Statusänderungen aus dem aktuellen Status und den Authorities des Nutzers.
+     *
+     * @param id vom Typ {@link UUID} um die Abfrage zu finden
+     * @return Liste von {@link TransitionModel} welche möglich sind
+     * @throws EntityNotFoundException falls die Abfrage nicht gefunden wird
+     */
+    public List<TransitionModel> getStatusAbfrageEventsBasedOnStateAndRole(final UUID id)
+        throws EntityNotFoundException {
+        List<AuthoritiesEnum> userRolesAuthorities = AuthenticationUtils.getUserAuthorities();
+        List<StatusAbfrageEvents> possibleAbfrageEventsBasedOnRole = new ArrayList<>();
+        Map<AuthoritiesEnum, StatusAbfrageEvents> rolesAndMatchingEvent = getRolesAndEventsMap();
+        List<TransitionModel> possibleTransitionsModel = new ArrayList<>();
+        for (AuthoritiesEnum userRole : userRolesAuthorities) {
```

Comment:
Evtl die For-Schleife in eine eigene Methode auslagern welche die möglichen AbfrageEvents basierend auf den Athorities extrahiert und dann die neue Methode hier aufrufen.

z.B. `getStatusAbfrageEventsForAuthorities(final List<AuthoritiesEnum> authorties)`